opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -209,11 +209,17 @@ namespace Opm
         } else {
             from = WellProducerCMode2String(ws.production_cmode);
         }
-        bool oscillating = std::count(this->well_control_log_.begin(), this->well_control_log_.end(), from) >= this->param_.max_number_of_well_switches_;
+        // Only keep first and last control until iterationIdx >= nupcol (These are used for output)
+        // After iteration >= nupcol we log all switches to check if the well controls oscillates
         const int episodeIdx = simulator.episodeIndex();
         const int iterationIdx = simulator.model().newtonMethod().numIterations();
         const int nupcol = schedule[episodeIdx].nupcol();
-        if (oscillating && iterationIdx > nupcol) {
+        if (iterationIdx < nupcol && this->well_control_log_.size() > 2) {
+            this->well_control_log_.erase (this->well_control_log_.begin()+1,this->well_control_log_.end()-1);
+        }
+
+        const bool oscillating = std::count(this->well_control_log_.begin(), this->well_control_log_.end(), from) >= this->param_.max_number_of_well_switches_;
+        if (oscillating) {
             // only output frist time
```

Comment:
not part of this PR, but since you are editing here.  `frist` should be first.  `bool output` can be `const`.